src/test/java/org/ethereum/beacon/discovery/DiscoveryManagerTest.java

```diff
@@ -45,6 +57,64 @@ public void testRegularHandshake() throws Exception {
     assertThat(pingRes).isCompleted();
   }
 
+  @Test
+  public void testInvalidHandshakeShouldDropsSession() throws Exception {
+    TestNetwork network = new TestNetwork();
+    TestManagerWrapper attackerNode = network.createDiscoveryManager(1);
+    TestManagerWrapper victimNode = network.createDiscoveryManager(2);
+
+    attackerNode.getDiscoveryManager().ping(victimNode.getNodeRecord());
+
+    TestMessage out1_1 = attackerNode.nextOutbound();
+    assertThat(out1_1.getPacket()).isInstanceOf(OrdinaryMessagePacket.class);
+    victimNode.deliver(out1_1); // Random (Ping is pending)
+
+    TestMessage out2_1 = victimNode.nextOutbound();
+    assertThat(out2_1.getPacket()).isInstanceOf(WhoAreYouPacket.class);
+    attackerNode.deliver(out2_1); // WhoAreYou
+
+    TestMessage validHandshakeMessage = attackerNode.nextOutbound(); // valid Handshake
+    assertThat(validHandshakeMessage.getPacket()).isInstanceOf(HandshakeMessagePacket.class);
+
+    // preparing Handshake with invalid id signature
+    NodeSession attackerToVictimSession =
+        attackerNode
+            .getDiscoveryManager()
+            .getNodeSession(victimNode.getNodeRecord().getNodeId())
+            .orElseThrow();
+    HandshakeMessagePacket handshakePacket =
+        (HandshakeMessagePacket) validHandshakeMessage.getPacket();
+    RawPacket handshakeRawPacket = validHandshakeMessage.getRawPacket();
+    Header<HandshakeAuthData> header = handshakePacket.getHeader();
+    Bytes invalidSignature = Functions.sign(attackerNode.getPrivateKey(), Bytes32.ZERO);
+    Header<HandshakeAuthData> malformedHeader =
+        Header.createHandshakeHeader(
+            header.getAuthData().getSourceNodeId(),
+            header.getStaticHeader().getNonce(),
+            invalidSignature,
+            header.getAuthData().getEphemeralPubKey(),
+            header.getAuthData().getNodeRecord(NodeRecordFactory.DEFAULT));
+    HandshakeMessagePacket malformedHandshakePacket =
+        HandshakeMessagePacket.create(
+            handshakeRawPacket.getMaskingIV(),
+            malformedHeader,
+            new PingMessage(Bytes.EMPTY, UInt64.ONE),
+            attackerToVictimSession.getInitiatorKey());
+    Bytes16 maskingKey = Bytes16.wrap(victimNode.getNodeRecord().getNodeId(), 0);
+    RawPacket malformedRawPacket =
+        RawPacket.createAndMask(
+            handshakeRawPacket.getMaskingIV(), malformedHandshakePacket, maskingKey);
+    TestMessage malformedMessage = attackerNode.createOutbound(malformedRawPacket, victimNode);
+
+    victimNode.deliver(malformedMessage); // Malformed Handshake
+    victimNode.deliver(malformedMessage); // Malformed Handshake
+    victimNode.deliver(validHandshakeMessage);
+
+    // victim node should drop the session on the first malformed handshake message
+    // and ignore any subsequent handshake messages
+    assertThatThrownBy(victimNode::nextOutbound);
```

Comment:
This is a little confusing to read without any assertions on the error thrown.  Should probably add some assertions here, or have `nextOutbound` just return an `Optional`.